dashboard/modules/train/train_head.py

```diff
@@ -92,6 +96,25 @@ async def get_train_runs(self, req: Request) -> Response:
             content_type="application/json",
         )
 
+    async def _add_actor_status(self, train_runs):
+        actor_status_table = {}
+        try:
+            logger.info("Getting all actor info from GCS.")
+            request = gcs_service_pb2.GetAllActorInfoRequest()
+            reply = await self._gcs_actor_info_stub.GetAllActorInfo(request, timeout=5)
+            if reply.status.code == 0:
+                for message in reply.actor_table_data:
+                    actor_table_data = actor_table_data_to_dict(message)
+                    actor_status_table[actor_table_data["actorId"]] = actor_table_data[
+                        "state"
+                    ]
+        except Exception:
+            logger.exception("Error Getting all actor info from GCS.")
+
+        for train_run in train_runs.values():
```

Comment:
for each train run, we should also check if the actor is dead and then mark the run status as something (maybe killed?).

This is for the case where the node dies before the state actor gets updated status information or something like that.

So train run status should be something like this:
if status is a terminal status like errored, or finished, use that status.
else check the actor status. If actor is alive, use the status.
else if actor is dead, mark it as killed or errored or something.

Reply:
I see, so we do the post-processing in TrainHead to ensure the running status is updated correctly. That makes sense